pallets/parachain-staking/src/lib.rs

```diff
@@ -1574,18 +1581,13 @@ pub mod pallet {
 								let mut collator = <CandidateInfo<T>>::get(&candidate_id)
 									.ok_or(Error::<T>::CandidateDNE)?;
 								T::Currency::unreserve(&delegator_id, balance_amt);
-								let before = collator.total_counted;
 								// need to go into decrease_delegation
 								let in_top = collator.decrease_delegation::<T>(
 									&candidate_id,
 									delegator_id.clone(),
 									amount_before,
 									balance_amt,
 								)?;
-								let after = collator.total_counted;
```

Comment:
Also whats the reason for removing this?

Reply:
Oh I see, you refactored this part into reset_top_delegations

Reply:
What about bottom? do they suffer from the same issue?

Actually here you are not doing a distinction on whether the collator is in top or not, you always called Pallet::<T>::update_active if the collator was active. Now you moved it into reset_top_data only, does not that change the beavior?

Reply:
> Also whats the reason for removing this?

The reason for removing it is that we put it in reset_top_data.

> Actually here you are not doing a distinction on whether the collator is in top or not, you always called Pallet::::update_active if the collator was active. Now you moved it into reset_top_data only, does not that change the beavior?

No because we only update total_counted if the top delegations total changes. If the bottom delegations total changes, it does not matter because bottom delegations are not counted.